app/models/region.rb

```diff
@@ -134,24 +134,35 @@ def dashboard_analytics(period:, prev_periods:, include_current_period: true)
   def syncable_patients
     case region_type
       when "block"
-        registered_patients.with_discarded
-          .or(assigned_patients.with_discarded)
-          .union(appointed_patients.with_discarded)
+        Patient.with_discarded.where(id: syncable_patient_ids)
       else
         registered_patients.with_discarded
     end
   end
 
-  def registered_patients
-    Patient.where(registration_facility: facility_regions.pluck(:source_id))
+  def syncable_patient_ids(exclude_facility: nil)
+    case region_type
+      when "block"
+        registered_patients(exclude_facility: exclude_facility)
+          .with_discarded.select(:id)
+          .or(assigned_patients(exclude_facility: exclude_facility).with_discarded.select(:id))
+          .union(appointed_patient_ids(exclude_facility: exclude_facility).with_discarded)
```

Comment:
This is not accurate, the `current_facility` exclusion only excludes registered patients in the current facility. Just line 146 should be it I think.

Reply:
Hrm. It's more complex than that, unfortunately. We need to exclude patients registered in the `current_facility` from all three groups :/. Thinking of a different solution now, since the appointment related changes are not viable.

Reply:
Oof yeah. Would using a `pluck` instead of a `select` at the top level work allow us to keep the top level `not` clause?